src/config.py

```diff
@@ -40,6 +40,12 @@ def get_logs_dir(self):
     def get_repos_dir(self):
         return self.config["STORAGE"]["REPOS_DIR"]
     
+    def get_openai_base_url(self):
+        return self.config["API_ENDPOINTS"]["OPENAI_BASE_URL"]
+    
+    def get_search_engine(self):
+        return self.config["USER_PREFS"]["SEARCH_ENGINE"]
```

Comment:
**Security Issue**: Storing configuration settings like the OpenAI base URL and user preferences in plain text within the application's configuration file can expose sensitive information. If the configuration file is accessed by an unauthorized user, they could potentially modify these settings to redirect traffic or alter the behavior of the application. <br> **Fix**: Encrypt sensitive configuration settings, such as API endpoints and user preferences, to protect them from unauthorized access and modifications. Additionally, consider implementing access controls to restrict who can read or modify the configuration file. <br> **Code Suggestion**: 
 ```
 +from cryptography.fernet import Fernet
 +
 +class SecureConfig:
 +    def __init__(self):
 +        self.key = Fernet.generate_key()
 +        self.cipher_suite = Fernet(self.key)
 +
 +    def encrypt_config(self, config_value):
 +        if isinstance(config_value, str):
 +            config_value = config_value.encode()
 +        encrypted_text = self.cipher_suite.encrypt(config_value)
 +        return encrypted_text
 +
 +    def decrypt_config(self, encrypted_value):
 +        decrypted_text = self.cipher_suite.decrypt(encrypted_value).decode('utf-8')
 +        return decrypted_text
 ```